test/FunctionalTests/Microsoft.OData.Core.Tests/Build.NetFramework/TestMetadataSelector.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Vocabularies;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Evaluation
+{
+    public class TestMetadataSelector : ODataMetadataSelector
+    {
+        public TestMetadataSelector()
+        {
+        }
+
+        public string PropertyToOmit;
+
+        public IEnumerable<IEdmNavigationProperty> NavigationPropertyToAdd;
+
+        /// <summary>
+        /// Determines which navigation links should be selected.
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="navigationProperties"></param>
+        /// <returns></returns>
+        public override IEnumerable<IEdmNavigationProperty> SelectNavigationProperties(IEdmStructuredType type, IEnumerable<IEdmNavigationProperty> navigationProperties)
+        {
+            if (NavigationPropertyToAdd != null)
+            {
+                return NavigationPropertyToAdd;

```

Comment:
>NavigationPropertyToAdd [](start = 23, length = 23)

So this returns only the NavigationpropertyToAdd, it does not add NavigationPropertyToAdd to the passed in navigationProperties, right?  Should it be instead named "NavigationPropertiesToReturn"?